### PR TITLE
HSEARCH-1955 ScheduledCommitPolicy helper threads not handling except…

### PR DESCRIPTION
**Pull request by Sanne**
…ions

 - https://hibernate.atlassian.net/browse/HSEARCH-1955

### CODE REVIEW

File: engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java [hunk 1]
```diff
@@ -80,6 +80,26 @@ public boolean evaluate() {
...
+			action = "throw new NullPointerException(\"Fake internal error\")",
+			name = "timerDisruptingError")
+	public void testErrorHandlingOnBackgroundThread() throws Exception {
+		writeData( sfAsyncExclusiveIndex, 2 );
+		final CountingErrorHandler errorHandler = (CountingErrorHandler) sfAsyncExclusiveIndex.getSearchFactory().getErrorHandler();
+		assertConditionMet( new Condition() {
+			@Override
+			public boolean evaluate() {
+				// It's going to commit once each millisecond, and produce a failure each time.
+				// So "4" is just a random number higher than 0, but high enough to
```
Comment (DavideD) [inline engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java]: Do you mean "2"?

Not critical, but maybe you can use a method or a constant do describe this better

Comment (DavideD) [inline engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java]: No need to change it.
I'll replace it with 2 before merging it

File: engine/src/main/java/org/hibernate/search/backend/impl/lucene/ExclusiveIndexWorkspaceImpl.java [hunk 2]
```diff
@@ -27,7 +28,8 @@ public ExclusiveIndexWorkspaceImpl(DirectoryBasedIndexManager indexManager, Work
 		boolean async = ! BackendFactory.isConfiguredAsSync( cfg );
 		if ( async ) {
 			int commitInterval = PropertiesParseHelper.extractFlushInterval( indexManager.getIndexName(), cfg );
-			commitPolicy = new ScheduledCommitPolicy( writerHolder, indexManager.getIndexName(), commitInterval );
+			ErrorHandler errorHandler = context.getErrorHandler();
+			commitPolicy = new ScheduledCommitPolicy( writerHolder, indexManager.getIndexName(), commitInterval, errorHandler );
 		}
 		else {
 			commitPolicy = new PerChangeSetCommitPolicy( writerHolder );
```
Comment (DavideD) [inline engine/src/main/java/org/hibernate/search/backend/impl/lucene/ExclusiveIndexWorkspaceImpl.java]: I suppose it's a different issue but shouldn't we handle errors for this case as well?

File: engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java (hunk 1 above)
Comment (Sanne) [inline engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java]: that's a rebase/conflict mistake. I actually meant to use "4" in the code, which is safer than 2 as t here are 2 updates being done, I want to verify the scheduler does more than that.

File: engine/src/main/java/org/hibernate/search/backend/impl/lucene/ExclusiveIndexWorkspaceImpl.java (hunk 2 above)
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/backend/impl/lucene/ExclusiveIndexWorkspaceImpl.java]: It's not needed for that one as it's not asynchronous. The problem we're fixing is that an exception would kill the timer thread, and never commit again.

File: engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java (hunk 1 above)
Comment (DavideD) [inline engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java]: I'll replace the threshold to 4 then

**Comment by Sanne** [comment]
pushed again to fix the '4' :)

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-search/pull/919

**Comment by Sanne** [comment]
I'll resolve the JIRA then

**Pull request closed without merging**